Quotient/room.cpp

```diff
@@ -1752,12 +1760,48 @@ Room::Private::moveEventsToTimeline(RoomEventsRange events,
         if (auto n = q->checkForNotifications(ti); n.type != Notification::None)
             notifications.insert(eId, n);
         Q_ASSERT(q->findInTimeline(eId)->event()->id() == eId);
+        updateThread(ti.event());
     }
     const auto insertedSize = (index - baseIndex) * placement;
     Q_ASSERT(insertedSize == int(events.size()));
     return Timeline::size_type(insertedSize);
 }
 
+void Room::Private::updateThread(const RoomEvent* event)
+{
+    const auto rme = eventCast<const RoomMessageEvent>(event);
+    if (rme == nullptr) {
+        return;
+    }
+    if (!rme->isThreaded()) {
+        return;
+    }
+
+    auto& thread = threads[rme->threadRootEventId()];
+    if (thread.threadRootId.isEmpty()) {
+        thread.threadRootId = rme->threadRootEventId();
+        const auto threadRootIndex = eventsIndex.constFind(rme->threadRootEventId());
+        // If we can't find the root we assume it's a historical event and will be loaded later.
+        if (threadRootIndex != eventsIndex.cend()) {
+            const auto rootEvent = timeline[Timeline::size_type(*threadRootIndex - q->minTimelineIndex())].viewAs<RoomMessageEvent>();
+            thread.addEvent(rootEvent, true, rootEvent->senderId() == connection->userId());
+        }
```

Comment:
Isn't all this essentially the same as
```suggestion
        // If we can't find the root we assume it's a historical event and will be loaded later.
        if (auto rootIt = q->findInTimeline(thread.threadRootId); rootIt != historyEdge())
            thread.addEvent(rootIt->viewAs<RoomMessageEvent>(), true,
                            (*rootIt)->senderId() == connection->userId());
```
or am I missing something? If I am, a comment in the code would be great to have.